components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -31,6 +31,25 @@ use ipc_channel::ipc::IpcSender;
 use servo_url::ServoUrl;
 use std::mem;
 
+#[derive(JSTraceable, MallocSizeOf)]
+#[unrooted_must_root_lint::must_root]
```

Comment:
The `must_root` attribute shouldn't be needed any more.

Reply:
Got it, I just rebased with latest and removed the attribute but I can't figure out why it complains telling me that
```
error: Expression of type canvas_state::CanvasState must be rooted
  --> components/script/dom/canvasrenderingcontext2d.rs:66:13
   |
66 |         let canvas_state =
   |             ^^^^^^^^^^^^
   |
   = note: `#[deny(unrooted_must_root)]` on by default

error: aborting due to previous error

error: could not compile `script`.
```

Any hints?

Reply:
Ah, that's unfortunate. I think the best thing to do here is split out the initialization of CanvasState into two parts - extract [this code](https://github.com/servo/servo/blob/3c426d707859091cae24a32db6575609c42fa646/components/script/canvas_state.rs#L151-L159) into a static method that returns a `RemoteCanvasState` struct, and make `CanvasState` accept a `RemoteCanvasState` argument. Then you can leave the `CanvasState::new` call where it was before this PR and avoid the rooting issues.

Reply:
Thanks josh! Now I can work on the next 26 dom structs!